hack/csi_client/cmd/main.go

```diff
@@ -1,34 +1,45 @@
 package main
 
 import (
+	"errors"
 	"flag"
 	"fmt"
+	"os"
 
-	csi "sigs.k8s.io/gcp-filestore-csi-driver/hack/csi_client/pkg/csi"
+	"sigs.k8s.io/gcp-filestore-csi-driver/hack/csi_client/pkg/csi"
 )
 
-const requestHelpMessage = `The type of CSI request to perform.
-Supported requests:
+const helpMessage = `
+Usage of %s:
+	The endpoint and request flags are required flags.
+	Other necessary flags are determined based on the request type.
+
 	NodePublish
-		Flags required: shareAddr, shareName, targetPath, smbUsername (WINDOWS), smbPassword (WINDOWS)
+		- shareAddr, shareName, targetPath, smbUsername (WINDOWS), smbPassword (WINDOWS)
+
 	NodeUnpublish
-		Flags required: targetPath`
+		- targetPath
 
-var (
-	endpoint    = flag.String("endpoint", "/tmp/csi.sock", "CSI endpoint")
-	shareAddr   = flag.String("shareAddr", "localhost", "Address of the share")
-	shareName   = flag.String("shareName", "", "Name of the share")
-	targetPath  = flag.String("targetPath", "", "Path to mount volume at")
-	smbUsername = flag.String("smbUsername", "", "WINDOWS ONLY: Username to login to SMB share")
-	smbPassword = flag.String("smbPassword", "", "WINDOWS ONLY: Password to login to SMB share")
+Supported flags:
+`
 
-	request = flag.String("request", "", requestHelpMessage)
+func main() {
+	flag.CommandLine = flag.NewFlagSet("test", flag.ExitOnError)
+	flag.ErrHelp = errors.New("flag: help req")
+	flag.CommandLine.Usage = func() {
```

Comment:
Is putting everything under `CommandLine` required? From the documentation, it seems like the variables are top-level in the pkg

Reply:
You're right, it doesn't. Didn't catch this earlier.